src/payment/models/crypto-route/crypto-route.service.ts

```diff
@@ -0,0 +1,118 @@
+import { CreateCryptoRouteDto } from './dto/create-crypto-route.dto';
+import { CryptoRoute } from './crypto-route.entity';
+import { DepositService } from '../deposit/deposit.service';
+import { RouteType } from '../route/deposit-route.entity';
+
+@Injectable()
+export class CryptoRouteService {
+  constructor(
+    private readonly cryptoRepo: CryptoRouteRepository,
+    private readonly assetService: AssetService,
+    private readonly stakingService: StakingService,
+    private readonly userService: UserService,
+    private readonly depositService: DepositService,
+  ) {}
+
+  // --- VOLUMES --- //
+  @Cron(CronExpression.EVERY_YEAR)
+  async resetAnnualVolumes(): Promise<void> {
+    this.cryptoRepo.update({ annualVolume: Not(0) }, { annualVolume: 0 });
+  }
+
+  async updateVolume(cryptoId: number, volume: number, annualVolume: number): Promise<void> {
+    await this.cryptoRepo.update(cryptoId, {
+      volume: Util.round(volume, Config.defaultVolumeDecimal),
+      annualVolume: Util.round(annualVolume, Config.defaultVolumeDecimal),
+    });
+
+    // update user volume
+    const { user } = await this.cryptoRepo.findOne({
+      where: { id: cryptoId },
+      relations: ['user'],
+      select: ['id', 'user'],
+    });
+    const userVolume = await this.getUserVolume(user.id);
+    await this.userService.updateCryptoVolume(user.id, userVolume.volume, userVolume.annualVolume);
+  }
+
+  async getUserVolume(userId: number): Promise<{ volume: number; annualVolume: number }> {
+    return this.cryptoRepo
+      .createQueryBuilder('crypto')
+      .select('SUM(volume)', 'volume')
+      .addSelect('SUM(annualVolume)', 'annualVolume')
+      .where('userId = :id', { id: userId })
+      .getRawOne<{ volume: number; annualVolume: number }>();
+  }
+
+  async getTotalVolume(): Promise<number> {
+    return this.cryptoRepo
+      .createQueryBuilder('crypto')
+      .select('SUM(volume)', 'volume')
+      .getRawOne<{ volume: number }>()
+      .then((r) => r.volume);
+  }
+
+  // --- CRYPTOS --- //
+  async createCrypto(userId: number, dto: CreateCryptoRouteDto): Promise<CryptoRoute> {
+    // check asset
+    const asset =
+      dto.buyType === BuyType.WALLET
+        ? await this.assetService.getAsset(dto.asset.id)
+        : await this.assetService.getAssetByDexName('DFI');
+    if (!asset) throw new BadRequestException('Asset not found');
+
+    // check staking
+    const staking =
+      dto.buyType === BuyType.STAKING ? await this.stakingService.getStaking(dto.staking.id, userId) : null;
+    if (dto.buyType === BuyType.STAKING && !staking) throw new BadRequestException('Staking route not found');
+
+    // check if exists
+    const existing = await this.cryptoRepo.findOne({
+      where: {
+        asset: dto.asset,
+        ...(dto.buyType === BuyType.WALLET ? { asset: asset, staking: IsNull() } : { staking: staking?.deposit }),
+        user: { id: userId },
+      },
```

Comment:
Should be possible to create a route per:
- User and
- Chain and
- Asset or Staking (target deposit)